subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -250,40 +246,96 @@ static int adv_send(struct bt_mesh_ext_adv *ext_adv, struct bt_mesh_adv *adv)
 	return err;
 }
 
-static const char * const adv_tag_to_str[] = {
-	[BT_MESH_ADV_TAG_LOCAL]  = "local adv",
-	[BT_MESH_ADV_TAG_RELAY]  = "relay adv",
-	[BT_MESH_ADV_TAG_PROXY]  = "proxy adv",
-	[BT_MESH_ADV_TAG_FRIEND] = "friend adv",
-	[BT_MESH_ADV_TAG_PROV]   = "prov adv",
-};
-
-static bool schedule_send_with_mask(struct bt_mesh_ext_adv *ext_adv, int ignore_mask)
+static int stop_proxy_adv(struct bt_mesh_ext_adv *ext_adv)
 {
-	if (atomic_test_and_clear_bit(ext_adv->flags, ADV_FLAG_PROXY)) {
-		atomic_clear_bit(ext_adv->flags, ADV_FLAG_PROXY_START);
-		(void)bt_le_ext_adv_stop(ext_adv->instance);
+	int err;
 
+	if (atomic_test_bit(ext_adv->flags, ADV_FLAG_PROXY)) {
+		err = bt_le_ext_adv_stop(ext_adv->instance);
+		if (err) {
+			LOG_ERR("Failed to stop proxy advertising: %d", err);
+			return err;
+		}
+
+		atomic_clear_bit(ext_adv->flags, ADV_FLAG_PROXY);
 		atomic_clear_bit(ext_adv->flags, ADV_FLAG_ACTIVE);
 	}
 
-	if (atomic_test_bit(ext_adv->flags, ADV_FLAG_ACTIVE)) {
-		atomic_set_bit(ext_adv->flags, ADV_FLAG_SCHEDULE_PENDING);
-		return false;
-	} else if ((~ignore_mask) & k_work_busy_get(&ext_adv->work)) {
-		return false;
+	return 0;
+}
+
+static int adv_queue_send_process(struct bt_mesh_ext_adv *ext_adv)
+{
+	struct bt_mesh_adv *adv;
+	int err = -ENOENT;
+
+	while ((adv = bt_mesh_adv_get_by_tag(ext_adv->tags, K_NO_WAIT))) {
+		/* busy == 0 means this was canceled */
+		if (!adv->ctx.busy) {
+			bt_mesh_adv_unref(adv);
+			continue;
+		}
+
+		if (!stop_proxy_adv(ext_adv)) {
```

Comment:
`stop_proxy_adv` now returns 0 on success. error check should be updated here and below.